### DISCUSSION TITLE
Adding custom directives

### DISCUSSION THREAD

**Discussion by nachtfunke** [nuxt/nuxt#18250] [Questions]
Hello! I'm fairly new to vue/nuxt - sorry if this is a dumb question. I am trying to add a custom directive. According to the nuxt3 docs and the vue3 docs, this is how it should be done (unless I just misunderstand):

#### directives.client.js

```jsx
import { defineNuxtPlugin } from '#app';

export default defineNuxtPlugin( nuxtApp => {
    nuxtApp.directive('visible', {
        mounted(el, binding) {
            el.style.visibility = !!binding.value ? 'visible':'hidden';
        }
    })
});
```

I am not doing anything else, because according to the docs, all the plugins are auto-discovered.

I don't get any special output or any react from the cli or in client console, but as soon as I am adding `v-visible` somewhere, the cli spits out:

```bash
[SSR] Error transforming /components/SiteHeader.vue: SyntaxError: Custom directive is missing corresponding SSR transform and will be ignored.
```

And in the client console, I see:

```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'deep')
    at withDirectives
```

Am I doing something wrong here, or did I stumble upon a bug?

**Comment by wobsoriano** [comment]
Hi, instead of `nuxtApp.directive`, can you do `nuxtApp.vueApp.directive`?

```ts
export default defineNuxtPlugin( nuxtApp => {
    nuxtApp.vueApp.directive('visible', {
        mounted(el, binding) {
            el.style.visibility = !!binding.value ? 'visible':'hidden';
        }
    })
});
```
